openwisp_monitoring/monitoring/static/monitoring/js/chart.js

```diff
@@ -16,6 +16,114 @@
         }
         return newArr;
     }
+
+    function adaptiveFilterPoints(charts, yrawVal) {
+
+        var convertToAdaptiveBytes = function(charts, type, i, multiplier, unit) {
+            charts[type].y[i] = Math.round((charts[type].y[i] * multiplier) * 100) / 100;
+            charts[type].hovertemplate[i] = charts[type].y[i] + ' ' + unit;
+        };
+
+        for (var i=0; i<charts[0].y.length; i++) {
+            if (yrawVal[i] == null) {
+                charts[0].hovertemplate[i] = 'N/A' + '<extra></extra>';
+            }
+            else if (charts[0].y[i] == 0) {
+                convertToAdaptiveBytes(charts, 0, i, 1, 'B');
+            }
+            else if (charts[0].y[i] < 0.001) {
+                convertToAdaptiveBytes(charts, 0, i, 1000000, 'KB');
+            }
+            else if (charts[0].y[i] < 1) {
+                convertToAdaptiveBytes(charts, 0, i, 1000, 'MB');
+            }
+            else {
+                convertToAdaptiveBytes(charts, 0, i, 1, 'GB');
+            }
+        }
+
+        for (i=0; i<charts[1].y.length; i++) {
+            if (yrawVal[i] == null) {
+                charts[1].hovertemplate[i] = 'N/A' + '<extra></extra>';
+            }
+            else if (charts[1].y[i] == 0) {
+                convertToAdaptiveBytes(charts, 1, i, 1, 'B');
+            }
+            else if (charts[1].y[i] < 0.001) {
+                convertToAdaptiveBytes(charts, 1, i, 1000000, 'KB');
+            }
+            else if (charts[1].y[i] < 1) {
+                convertToAdaptiveBytes(charts, 1, i, 1000, 'MB');
+            }
+            else {
+                convertToAdaptiveBytes(charts, 1, i, 1, 'GB');
+            }
+        }
+
+        for (i=0; i<charts[2].y.length; i++) {
+            if (yrawVal[i] == null) {
+                charts[2].hovertemplate[i] = 'N/A' + '<extra></extra>';
+            }
+            else if (charts[2].y[i] == 0) {
+                convertToAdaptiveBytes(charts, 2, i, 1, 'B');
+            }
+            else if (charts[2].y[i] < 0.001) {
+                convertToAdaptiveBytes(charts, 2, i, 1000000, 'KB');
+            }
+            else if (charts[2].y[i] < 1) {
+                convertToAdaptiveBytes(charts, 2, i, 1000, 'MB');
+            }
+            else {
+                convertToAdaptiveBytes(charts, 2, i, 1, 'GB');
+            }
+        }
+    }
+
+    function adaptiveFilterLayout(charts, layout) {
+        var newArr = charts[0].y, sum = 0, count = 0;
```

Comment:
I don't think `newArr` is a good variable name here, because it's not really a new array but it's the same array which we are reading. I'd call it simply `y`.